### DISCUSSION TITLE
Ambient display not working but random colors are shown

### DISCUSSION THREAD

**Discussion by shariq-azim** [awawa-dev/HyperHDR#615] [Q&A]
Hello i am confused why would ws2812b not show true ambient color of laptops screen but random color patterns? I am running v19 of hyperhdr

Here is the connections in subsequent images.. yellow of psu(20amps 5v) [V+]  goes to ws2812b red
Brown goes from psu(V-) to led white
Blue goes from psu (V-) to raspberry pi ground

The blue from psu (v-) goes to ground of raspberry pi.
Green goes from gpio 18(6th pin from the top right) of raspberry 3b to data line (green) of ws2812b

And from the first two images.

[https://imgur.com/RxxNwzL](url)
[https://imgur.com/dVmrDWq](url)
[https://imgur.com/aREsMhZ](url)
[https://imgur.com/4RloAwU](url)
video [https://drive.google.com/file/d/1QifGly-0HhFhdfMVe9YZyZqng999hAx7/view?usp=drive_link](url)
all related images in a folder [https://drive.google.com/drive/folders/1RJAekurwSqNgljMUvOyZHA_aWdY1dunv?usp=sharing](url)

Yellow from psu (v+) goes to red of led
Brown from psu(v-) goes to white of led
Green of raspberry pi goes to green of led.

i have also attached a simple connection where the additional wires are directly connected to the red and white wires of ws2812b

Please help im annoyed for past 2 weeks

**Comment by awawa-dev** [comment]
Hi
Just to remind FAQ and our official position: https://github.com/awawa-dev/HyperHDR/wiki/FAQ There is an example of the configuration you try to use which requires root privilages. Recommended way (in fact, the easiest and also meeting the LED technical parameters) is described detailed here: https://www.hyperhdr.eu/2023/02/ultimate-guide-on-how-to-build-led.html because you are missing the 3.3v to 5v level shifter too.

I think you should attach the HyperHDR logs (full!) to see whats going on with your LED driver and screenshot of your HyperHDR LED driver configuration page. Try also to set solid color instead of the effect for example red or green and see what's the result (still random?).

**Reply by shariq-azim** [reply]
Hello @awawa-dev ,

As per your comments i have brought  a level shifter and placed in these configuration


[https://drive.google.com/file/d/1eKrD0zowMmGIWnDD74KgmeKJSlphzx94/view?usp=drive_link](url)
[https://drive.google.com/file/d/1FyLtdFvIjV9h17JCPR3_8QipB2rilpF1/view?usp=drive_link](url)

But now the disco effect is gone however it now run only **10** leds


Please find the logs

2023-07-29T11:34:34.625Z [WEBSOCKET] (WebSocketClient.cpp:30) New connection from ::ffff:192.168.0.6
2023-07-29T11:34:34.628Z [WEBSOCKET] (JsonAPI.cpp:97) Client '::ffff:192.168.0.6' switch to HyperHDR instance 0
2023-07-29T11:34:34.949Z [DB] Database opened: /.hyperhdr/db/hyperhdr.db
2023-07-29T11:34:37.020Z [IMAGETOLED0] Total index number is: 19648 (memory: 19648). User sparse processing is: disabled, image size: 320 x 240, area number: 60
2023-07-29T11:34:38.979Z [WEBSOCKET] (JsonAPI.cpp:1250) log streaming activated for client ::ffff:192.168.0.6
2023-07-29T11:35:29.708Z [V4L2:FY HD VIDEO: F] Video cache: enabled, size: 1, hits: 1155, needed: 1, cleanup: 0, limit: 8
2023-07-29T11:35:29.713Z [PERFORMANCE] [USB capturing: FPS = 10.00, decoding = 2ms, frames = 601, invalid = 0], [INSTANCE0: FPS = 10.00, processed = 601], [LED0: FPS = 10.00, send = 600, processed = 3000, dropped = 0]
2023-07-29T11:36:29.705Z [PERFORMANCE] [USB capturing: FPS = 10.00, decoding = 3ms, frames = 600, invalid = 0], [INSTANCE0: FPS = 10.00, processed = 600], [LED0: FPS = 10.00, send = 600, processed = 3000, dropped = 0]
2023-07-29T11:37:28.196Z [IMAGETOLED0] Total index number is: 19840 (memory: 19840). User sparse processing is: disabled, image size: 320 x 240, area number: 60
2023-07-29T11:37:29.697Z [PERFORMANCE] [USB capturing: FPS = 10.00, decoding = 3ms, frames = 600, invalid = 0], [INSTANCE0: FPS = 10.00, processed = 600], [LED0: FPS = 10.00, send = 600, processed = 3000, dropped = 0]
2023-07-29T11:37:37.396Z [IMAGETOLED0] Total index number is: 19648 (memory: 19648). User sparse processing is: disabled, image size: 320 x 240, area number: 60
2023-07-29T11:38:29.689Z [PERFORMANCE] [USB capturing: FPS = 10.00, decoding = 3ms, frames = 600, invalid = 0], [INSTANCE0: FPS = 10.00, processed = 600], [LED0: FPS = 10.00, send = 600, processed = 3000, dropped = 0]
2023-07-29T11:39:29.685Z [PERFORMANCE] [USB capturing: FPS = 10.00, decoding = 3ms, frames = 600, invalid = 0], [INSTANCE0: FPS = 10.00, processed = 600], [LED0: FPS = 10.00, send = 600, processed = 3000, dropped = 0]


Where am i doing wrong?

Regards,
Shariq

**Reply by awawa-dev** [reply]
The level shifter you used is not recommended for HyperHDR, designed for I2C are usually too slow for LEDs and work especially poorly with PWM Neopixel protocol (more likely to work with a very slow LED SPI bus). Anyway no guarantee at all...may work, may work partially or at all.

`I think you should attach the HyperHDR logs (full!) to see whats going on with your LED driver and screenshot of your HyperHDR LED driver configuration page.`

**Reply by awawa-dev** [reply]
BTW make sure you have default log level settings set in the logs tab (needs restart):
![obraz](https://github.com/awawa-dev/HyperHDR/assets/69086569/f773d499-5ed7-40e3-9b31-f1170a44a500)

**Reply by shariq-azim** [reply]
Hello,

The configs are in the dgrive image



video config https://drive.google.com/file/d/1mlS1lCJFeD8W3hoEUbK1Zgv95l4mTLXe/view?usp=drive_link

effect https://drive.google.com/file/d/1SI9DG8uoMOPiKU6og2KZp7s0DkyPFveY/view?usp=drive_link
layout https://drive.google.com/file/d/14hxX07zGKi9f2jBL3h36eWSbYRLKseoP/view?usp=drive_link

please find the logs
https://drive.google.com/file/d/1B7HSEX7ksLP6DizVHWyUXXP0HIGADyTF/view?usp=drive_link

Btw, now leds wont even run a single led. Am i doing something wrong with layout?

Regards,
Shariq

**Reply by awawa-dev** [reply]
We need a screenshot of the LED driver setup. You seem to have now set GPIO15 to output there. And it doesn't work for sure.
```
[{"colorOrder":"bgr","currentLedCount":32,"dma":5,"gpio":15,"invert":false,"leds":60,"pwmchannel":0,"refreshTime":100,"rgbw":false,"type":"ws281x","whiteAlgorithm":"subtract_minimum"}]
2023-07-29T12:41:50.601Z [LEDDEVICE_WS281X] (LedDevice.cpp:103) Enable device
2023-07-29T12:41:50.602Z [LEDDEVICE_WS281X] Device disabled, device 'ws281x' signals error: 'Failed to open. Error message: Selected GPIO not possible'
```
Why dont you follow the link from the FAQ? https://github.com/awawa-dev/HyperHDR/discussions/111 You can try to set refreshTime to 0 to have better responsiveness.

**Reply by shariq-azim** [reply]
sorry for the confusion i was testing against  gpio 15 for a while
here is  the led config page(if i understood it correctly)
![image](https://github.com/awawa-dev/HyperHDR/assets/37748914/cf707131-71ec-4e9b-a74c-eda8872b28a8)
I have also set the refresh time to 0 

i am checking the post now i dont know how to make the most from it.

**Reply by awawa-dev** [reply]
RGB order is set differently but it may depend on the particular LED strip type. Does you LED work now? Please post current logs.

**Reply by shariq-azim** [reply]
i wish i could say yes, btw now only 10 leds are working
here is the logs
[logs 2.txt](https://github.com/awawa-dev/HyperHDR/files/12207374/logs.2.txt)
Btw, even the callibration mode is not working

**Reply by awawa-dev** [reply]
The current LED driver configuration is missing from the logs: try restarting HyperHDR. But now I don't see any errors. Driving the LEDs directly from the Rpi sometimes causes similar side effects, but usually with more LEDs when the Rpi doesn't meet the strict timing requirements of the Neopixel and the last part of the LED strip may not light up. Or a non-recommended I2C level shifter? Hard to say: both are not officially supported by HyperHDR. Calibration mode certainly works, but it's not forgiving of not meeting the required conditions: best used with Firefox.

**Reply by shariq-azim** [reply]
the callibration part i meant, that there should be solid color on the led when i the color was shown. but that doesnt works. not sure what firefox will do here. sorry for the confusion.
Btw, i have notice when  raspberry pi starts it gives the booting effect from hyperion. now my question is the voltag,e from gpio 18 was an issue then why would the boot effect still show?

also, i had a spare nodemcu where i installed wled and it gave me all the effects and colors without issues, so my led is is in working condition thats for sure

**Reply by awawa-dev** [reply]
esp or Raspberry Pi Pico boards  have specialized and reliable hardware acceleration that allows for reliable operation of many types of LEDs and cannot be compared to standard Raspberry Pi unit. It's hard to tell from your description what your problem is. try use 'identify button' when you right click on particular LED if it works: https://github.com/awawa-dev/HyperHDR/pull/379
![obraz](https://github.com/awawa-dev/HyperHDR/assets/69086569/f3f40c2b-65d0-45c0-9fbd-a98580da4513)
it should blink your LED so you can test if the last segment is working or not.

**Comment by shariq-azim** [comment]
alright , so i made it work installing wled as a controller running on nodemcu esp8266.
i know this isnt the solution to this original post, but for me its better than the frustrations. here is the video of it

https://github.com/awawa-dev/HyperHDR/assets/37748914/48fcbafe-5ea5-4f31-87c5-9e42b075d741

https://drive.google.com/file/d/1bfKFijEubYBJiUOcWUU_vcNlr-ilUk7t/view

Any tips to reduce the lags and make it more responsive?

**Reply by awawa-dev** [reply]
Read performance stats on your main HyperHDR page.

**Reply by awawa-dev** [reply]
And full logs of course.

**Reply by awawa-dev** [reply]
in old logs you had very low framerate 20FPS (640 x 480 @ 20 fps yuyv) which is definitely not good. Read  more: https://github.com/awawa-dev/HyperHDR/wiki/Quick-start#performance-pro-tips
